src/sdk/config_fs.js

```diff
@@ -151,88 +218,184 @@ class ConfigFS {
+     * get_account_relative_path_by_id returns the full account path by id
+     * the target of symlinks will be the 
+     * @param {string} account_id
+     * @returns {string} 
+    */
+    get_account_relative_path_by_id(account_id) {
+       return path.join('../', CONFIG_SUBDIRS.IDENTITIES, account_id, this.json('identity'));
+    }
+
+    /**
+     * get_account_or_user_path_by_id returns the full account/user path by id as following
+     * {config_dir}/identities/{account_id}/identity.json
+     * @param {string} account_id
+     * @returns {string} 
+    */
+    get_identity_path_by_id(account_id) {
+        return path.join(this.identities_dir_path, account_id, this.json('identity'));
+    }
+
+    /**
+     * get_identity_by_id returns the full account/user data by id from the following path
+     * {config_dir}/identities/{account_id}/identity.json
+     * @param {string} id
+     * @param {string} [type]
+     * @param {{show_secrets?: boolean, decrypt_secret_key?: boolean, silent_if_missing?: boolean}} [options]
+     * @returns {Promise<Object>} 
+    */
+    async get_identity_by_id(id, type, options = {}) {
+        const account_path = this.get_identity_path_by_id(id);
+        let identity = await this.get_identity_config_data(account_path, { ...options, silent_if_missing: true });
+
+        if (!identity && type === TYPES.ACCOUNT) {
+            identity = await this.search_accounts_by_id(id, options);
+        }
+        if (!identity && !options.silent_if_missing) {
+            const err = new Error(`Could not find identity by id ${id}`);
+            err.code = 'ENOENT';
+            throw err;
+        }
+        return identity;
+    }
+
+    /**
+     * search_accounts_by_id searches all accounts and finds an account that its _id matches the given id param
+     * @param {string} id
+     * @param {{show_secrets?: boolean, decrypt_secret_key?: boolean, silent_if_missing?: boolean}} [options]
+     * @returns {Promise<Object>} 
+    */
+    async search_accounts_by_id(id, options = {}) {
+        const old_account_names = await this.list_accounts();
+        for (const account_name of old_account_names) {
+            const account_data = await this.get_account_by_name(account_name, { ...options, silent_if_missing: true });
+            if (account_data._id === id) return account_data;
+        }
+    }
+
+    /**
+     * get_identities_by_id returns the full account/user data by id from the following path
+     * {config_dir}/identities/{id}/identity.json
+     * @param {string[]} ids
+     * @returns {Promise<Object>} 
+    */
+    async get_identities_by_id(ids, options = {}) {
+        const res = [];
+        for (const id of ids) {
+            const id_path = this.get_identity_path_by_id(id);
+            const id_data = await this.get_identity_config_data(id_path, { ...options, silent_if_missing: true });
+            res.push(id_data);
```

Comment:
Since `silent_if_missing: true` `id_data` can be undefined, do you want to push undefined?
If so, please add a comment.